letsencrypt-auto-source/letsencrypt-auto

```diff
@@ -152,10 +152,11 @@ su_sudo() {
   su root -c "$args"
 }
 
-if [ "$1" = "--cb-auto-has-root" ]; then
-  shift 1
-elif [ "$1" != "--le-auto-phase2" ]; then
-  # if $1 is --le-auto-phase2, we've executed this branch before
+# Sets the environment variable SUDO to be the name of the program or function
+# to call to get root access. If this script already has root privleges, SUDO
+# is set to an empty string. The value in SUDO should be run with the command
+# to called with root privileges as arguments.
+set_root_auth_mechanism() {
```

Comment:
I didn't change `say` or `error` as they're used all over the place and otherwise are unaffected by this PR, but I believe all other names now use the same convention and I created #5012 to track finishing this up.